tsdb/head.go

```diff
@@ -119,9 +122,12 @@ type HeadOptions struct {
 	// StripeSize sets the number of entries in the hash map, it must be a power of 2.
 	// A larger StripeSize will allocate more memory up-front, but will increase performance when handling a large number of series.
 	// A smaller StripeSize reduces the memory allocated, but can decrease performance with large number of series.
-	StripeSize     int
-	SeriesCallback SeriesLifecycleCallback
-	NumExemplars   int
+	StripeSize            int
+	SeriesCallback        SeriesLifecycleCallback
+	EnableExemplarStorage bool
+
+	// Runtime reloadable options.
+	MaxExemplars int
```

Comment:
Some late thought: This should be a atomic variable because it can be read and written to in parallel. Although it won't cause issues in real world, it can cause some flaky races in tests.